src/services/codefixes/fixImportNonExportedMember.ts

```diff
@@ -0,0 +1,114 @@
+/* @internal */
+namespace ts.codefix {
+    const fixId = "importNonExportedMember";
+    const errorCodes = [
+        Diagnostics.Module_0_declares_1_locally_but_it_is_not_exported.code
+    ];
+    registerCodeFix({
+        errorCodes,
+        getCodeActions(context) {
+            const { sourceFile } = context;
+            const info = getInfo(sourceFile, context, context.span.start);
+            if (!info || info.originSourceFile.isDeclarationFile) {
+                return undefined;
+            }
+            const changes = textChanges.ChangeTracker.with(context, t => doChange(t, info.originSourceFile, info.node));
+            return [createCodeFixAction(fixId, changes, [Diagnostics.Export_0_from_module_1, info.node.text, showModuleSpecifier(info.importDecl)], fixId, Diagnostics.Add_all_missing_exports)];
+        },
+        fixIds: [fixId],
+        getAllCodeActions: context => codeFixAll(context, errorCodes, (changes, diag) => {
+            const info = getInfo(diag.file, context, diag.start);
+            if (info) doChange(changes, info.originSourceFile, info.node);
+        }),
+    });
+
+    interface Info {
+        readonly node: Identifier;
+        readonly importDecl: ImportDeclaration;
+        readonly originSourceFile: SourceFile
+    }
+    function isImportDeclaration(node: Node): node is ImportDeclaration {
+        return node.kind === SyntaxKind.ImportDeclaration;
+    }
+
+    function getInfo(sourceFile: SourceFile, context: CodeFixContext | CodeFixAllContext, pos: number): Info | undefined {
+        const node = getTokenAtPosition(sourceFile, pos);
+        if (node && isIdentifier(node)) {
+            const importDecl = findAncestor(node, isImportDeclaration);
+            if (!importDecl || !isStringLiteralLike(importDecl.moduleSpecifier)) {
+                return undefined;
+            }
+            const resolvedModule = getResolvedModule(sourceFile, importDecl.moduleSpecifier.text);
+            const originSourceFile = resolvedModule && context.program.getSourceFile(resolvedModule.resolvedFileName);
+            if (!originSourceFile) {
+                return undefined;
+            }
+            return { node, importDecl, originSourceFile };
+        }
+    }
+
+    function getNamedExportDeclaration(sourceFile: SourceFile): ExportDeclaration | undefined {
+        let namedExport;
+        const statements = sourceFile.statements.filter(isExportDeclaration);
+        for (const statement of statements) {
+            if (statement.exportClause && isNamedExports(statement.exportClause)) {

```

Comment:
Avoid the intermediate array

```suggestion
        for (const statement of sourceFile.statements) {
            if (isExportDeclaration(statement) && statement.exportClause && isNamedExports(statement.exportClause)) {

```